### DISCUSSION TITLE
My Pico W is halting without a "lifeline" to Thonny

### DISCUSSION THREAD

**Discussion by slugvision** [micropython/micropython#11643] [RP2040 / Pico]
RP Pico W is very easy to use with micropython and Thonny. I have made a number of different programs without problems. I flash the programs to the Pico and then run them. When disconneting the pico from Thonny by closing Thonny down, the program (main.py) then executes automtically as expected. However, a particular program does not execute correctly, unless I keep Thonny connected (it is showing the micropython prompt). So, in other words, I flash the program to the pico, checks that it works (as main.py on the pico). But when powering from a USB power supply or just connected to the PC with Thonny shut down, the program halts!

I am executing some simple string operations on a file downloaded from the internet via the urequests. The problem seems to be the string operation, which is  very simple, like "mystr=string_record[1:10]". Again it also works and generates the correct result, when the program is executing with Thonny conneceted, still running on the target pico w. I have tried to rearrange the various codes part, but when adding print before and after this substring operation, I have identified that this string operation seems to be the culprit. But why? - hiw come the program is working correctly when Thonny is connected?  I have used standard os builds for the pico w, and I have also made my own os builds. Again, it is not a problem to get the program executing correctly when starting the program in Thonny, but when main.py is just started, when powering up without Thonny, it stops. What could be the problem?

**Comment by robert-hh** [comment]
Instead of using Thonny you should connect to the board using a simple terminal emulator like mpremote or picocom or putty. Then you will get error messages which can tell you what is missing. Thonny does a lot of additional set up operations, which are missing if you do not use Thonny.

**Reply by slugvision** [reply]
Thanks for the hint, I will give it a try..

**Comment by bixb922** [comment]
I suggest to connect to the RP with mpremote to see what message shows up there

**Reply by slugvision** [reply]
Thanks for the hint, I will give it a try..
